docs/operators/transform.md

```diff
@@ -0,0 +1,137 @@
+Transform - Operator Documentation
+==================================
+
+About Transform operator
+------------------------
+
+Transform means mapping of field expression from input to output or conversion of fields from one type to another.
+This operator is stateless. This operator receives objects on its input port; for each such input object, it creates a new output object whose fields are computed as expressions involving fields of the input object. 
+The types of the input and output objects are configurable as are the expressions used to compute the output fields. 
+
+The operator class is TransformOperator located in the package com.datatorrent.lib.transform.
+
+Use Case
+--------
+
+Consider the data received from input source and want to transform the data as below output schema while emitting to downstream operator. 
+
+Consider input objects with these fields:
+
+`{FirstName, LastName, Phone, DateOfBirth, Address}`
+ 
+ and output objects with fields: 
+ 
+ `{Name, Phone, Age, Address}`. 
+ 
+Suppose Name is a concatenation of FirstName and LastName and 
+        Age is computed by subtracting the DateOfBirth from the current year.
+
+These simple computations can be expressed as Java expressions where the input object is
+represented by $ and provided as configuration parameters as follows:
+
+```
+Name => {$.FirstName}.concat(\" \").concat({$.LastName})
+Age => (new java.util.Date()).getYear() - {$.dateOfBirth}.getYear()
+```
+
+#### Configuration Parameters
+-----------------------------
+
+-   ***expressionMap*** -   Map<String, String>
+    -   Mandatory Parameter
+    -   Specifies the map between the output field and expression using fields of input Java object.
```

Comment:
Suggest rewriting for better clarity:
Specifies the map between the output field (key) and the expression used to compute it (value) using fields of the input Java object.